Std/Data/String.lean

```diff
@@ -0,0 +1,19 @@
+/-
+Copyright (c) 2022 Jannis Limperg. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jannis Limperg
+-/
+
+namespace String
+
+/--
+If `pre` is a prefix of `s`, i.e. `s = pre ++ t`, return the remainder `t`.
+-/
+def dropPrefix? (s : String) (pre : String) : Option Substring :=
+  let s := s.toSubstring
+  if s.take pre.length == pre.toSubstring then
+    s.drop pre.length
+  else
+    none
```

Comment:
This looks like it could easily be written to take `Substring` arguments as well. How about having both versions and implement the string version in terms of the substring version?

Reply:
Done. Wasn't super easy, but the function is more efficient as a result.